geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/OrderStatisticsTree.java

```diff
@@ -205,6 +205,32 @@ public boolean add(E element) {
     return true;
   }
 
+  public ArrayList<E> getIndexRange(int min, int max) {
+    ArrayList<E> entryList = new ArrayList<>();
+    Node<E> current = getNode(min);
+    for (int i = min; current != null && i <= max; i++) {
+      entryList.add(current.key);
```

Comment:
This could have a serious memory impact if the given range is large - compounded since the caller (`zrange`) ends up creating another List. Probably would be better to just inline this method.

Reply:
Hmm. The problem is that here, the OrderStatisticsTree only has OrderedSetEntries. So we're getting a list of OrderedSetEntries out of OrderStatisticsTree, and then in RedisSortedSet.zrange() we're returning a list of byte[]'s. The bytes have to be extracted _somehow_. I'd like to hold off and see how we implement a couple of the other range commands, to see what the best way to refactor this is.

Reply:
OK - I just wouldn't want that to get lost. Another option is to return an `Iterator` here. It seems fairly trivial to augment the internal `TreeIterator` with a 'limit'. Anyway, your call.

Reply:
An iterator's an excellent idea; I've pushed up an implementation!

Reply:
Nice. I like it!